django/forms/widgets.py

```diff
@@ -689,9 +690,22 @@ def create_option(
         self, name, value, label, selected, index, subindex=None, attrs=None
     ):
         index = str(index) if subindex is None else "%s_%s" % (index, subindex)
-        option_attrs = (
-            self.build_attrs(self.attrs, attrs) if self.option_inherits_attrs else {}
-        )
+
+        if not hasattr(self, "option_attrs"):
+            # AutocompleteSelect subclasses AutocompleteMixin first, and then Select
+            # (which subclasses this class).
+            # So the ChoiceWidget's __init__ method (which is how other Select classes
+            # get their option_attrs set) is not called.
+            # But AutocompleteSelect does call create_option, so we need to allow for
+            # a missing option_attrs attribute.
```

Comment:
All this can be simplified to just say that subclasses have mixins which do not call `super().__init__()` but do call `create_option()` … or something to that effect… :)

Reply:
To nessita, felix or carlton: can we not make `AutocompleteMixin` just honour the contract of `__init__()`? It's rare to have a mixin break the contract like this and I feel like if subclasses & mixins want to do this then they must behave properly.